dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -157,6 +163,34 @@ public Plan(ISKFunction function)
         this.Outputs = outputs;
         this._steps.Clear();
         this.AddSteps(steps.ToArray());
+        this.ConfigureTelemetry();
+    }
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+
+    /// <summary>
+    /// Configure telemetry resources:
+    ///     1. Meters.
+    ///     2. Activity Source.
+    /// </summary>
+    private void ConfigureTelemetry()
```

Comment:
As described above, I think all metering tools should be implemented as `static` and initialized in place of definition. Otherwise, this method forces us to call it every time when we define new constructor, which shouldn't be the case.